src/PrestaShopBundle/Form/Admin/Sell/Product/ExtraModulesType.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShop\PrestaShop\Adapter\Module\ModuleDataProvider;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShop\PrestaShop\Core\Module\ModuleRepository;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * This form type is used to display modules in an extra tab that regroup the modules implementing the displayAdminProductsExtra
+ * hook. This is not the recommended way to integrate in the product page anymore but we keep it for backward compatibility.
+ */
+class ExtraModulesType extends TranslatorAwareType
+{
+    public const HOOK_NAME = 'displayAdminProductsExtra';
```

Comment:
Nothing to change here, just wonder if we should pursue to use constants everywhere where using hooks (and why haven't we tried to do so before? like if we would have some static class to hold all the hooks or so?) or it sounds bad?
There are so many "magic strings" hanging now (i mean the hook names) everywhere :man_shrugging:

Reply:
Yeah probably for the core hooks at least it could make sense to have some consts, but it's not the purpose of this PR to refacto everything 😅 And it won't be applicable everywhere because we also have many dynamic hooks that are built with placeholders such as the object class names
But it's an idea to look into for future versions I guess